ethexe/contracts/src/MirrorAbi.sol

```diff
@@ -0,0 +1,177 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.26;
+
+import {Proxy} from "@openzeppelin/contracts/proxy/Proxy.sol";
+import {IMirror} from "./IMirror.sol";
+import {Gear} from "./libraries/Gear.sol";
+import {IRouter} from "./IRouter.sol";
+import {IWrappedVara} from "./IWrappedVara.sol";
+
+contract MirrorAbi is IMirror, Proxy {
+    bytes32 public stateHash;
+    address public inheritor;
+    /// @dev This nonce is the source for message ids unique generations.
+    /// Must be bumped on each send.
+    /// Zeroed nonce is always represent init message by eligible account.
+    uint256 public nonce;
+    address public router;
+    address public initializer;
+    address public impl;
+    address private _abi;
+    bytes32 private constant PROXY_SLOT = 0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc;
+
+    function initialize(address _initializer, address _router, address _impl, address _interface) public {
+        require(initializer == address(0), "initializer could only be set once");
+
+        initializer = _initializer;
+        router = _router;
+        impl = _impl;
+        _abi = _interface;
+
+        assembly {
+            sstore(PROXY_SLOT, _interface)
+        }
+    }
+
+    function _delegate(address) internal override {
```

Comment:
```suggestion
    function _delegate(address _implementation) internal override {
```

Reply:
Why is that necessary?

Reply:
best practice

Reply:
But this triggers a warning
`Unused function parameter. Remove or comment out the variable name to silence this warning.`